samza-core/src/main/java/org/apache/samza/table/remote/RemoteTableProvider.java

```diff
@@ -83,11 +91,33 @@ public Table getTable() {
     TableRateLimiter.CreditFunction<?, ?> writeCreditFn;
     TableRateLimiter writeRateLimiter = null;
 
+    TableRetryPolicy readRetryPolicy = deserializeObject(READ_RETRY_POLICY);
+    TableRetryPolicy writeRetryPolicy = null;
+
+    if (readRetryPolicy != null || writeRetryPolicy != null) {
+      retryExecutor = Executors.newSingleThreadScheduledExecutor(runnable -> {
```

Comment:
Wondering if it's better to share this across all tasks and tables?

Reply:
Makes sense. I'll make this a singleton